cfme/storage/volume.py

```diff
@@ -0,0 +1,169 @@
+
+    @View.nested
+    class relationships(Accordion):  # noqa
+        tree = ManageIQTree()
+
+
+class VolumeView(BaseLoggedInPage):
+    """Base class for header and nav check"""
+    @property
+    def in_volume(self):
+        nav = ['Storage', 'Volumes'] if self.context['object'].appliance.version < '5.8' else \
+            ['Storage', 'Block Storage', 'Volumes']
+        return (
+            self.logged_in_as_current_user and
+            self.navigation.currently_selected == nav and
+            match_location(controller='cloud_volume', title='Cloud Volumes'))
+
+
+class VolumeAllView(VolumeView):
+    @property
+    def is_displayed(self):
+        return (
+            self.in_volume and
+            self.entities.title.text == 'Cloud Volumes')
+
+    toolbar = View.nested(VolumeToolbar)
+    entities = View.nested(VolumeEntities)
+    paginator = View.nested(PaginationPane)
+
+
+class VolumeDetailsView(VolumeView):
+    @property
+    def is_displayed(self):
+        expected_title = '{} (Summary)'.format(self.context['object'].name)
+        return (
+            self.in_volume and
+            self.entities.title.text == expected_title and
+            self.entities.breadcrumb.active_location == expected_title and
+            self.entities.relationships.get_text_of('Parent Cloud Provider') == self.context[
+                'object'].provider.name)
+
+    toolbar = View.nested(VolumeDetailsToolbar)
+    sidebar = View.nested(VolumeDetailsAccordion)
+    entities = View.nested(VolumeDetailsEntities)
+
+
+class VolumeAddEntities(View):
+    breadcrumb = BreadCrumb()
+    title = Text('//div[@id="main-content"]//h1')
+
+
+class VolumeAddForm(View):
+    volume_name = TextInput(name='name')
+    size = TextInput(name='size')
+    tenant = BootstrapSelect(id='cloud_tenant_id')
+    add = Button('Add')
+    cancel = Button('Cancel')
+
+
+class VolumeAddView(VolumeView):
+    @property
+    def is_displayed(self):
+        expected_title = "Add New Cloud Volume"
+        return (
+            self.in_volume and
+            self.entities.title.text == expected_title and
+            self.entities.breadcrumb.active_location == expected_title)
+
+    entities = View.nested(VolumeAddEntities)
+    form = View.nested(VolumeAddForm)
+
+
+class Volume(Navigatable):
+    def __init__(self, name, provider, appliance=None):
+        Navigatable.__init__(self, appliance=appliance)
+        self.name = name
+        # TODO add storage provider parameter, needed for accurate details nav
+        # the storage providers have different names then cloud providers
+        # https://bugzilla.redhat.com/show_bug.cgi?id=1455270
+        self.provider = provider
+
+
+@navigator.register(Volume, 'All')
+class VolumeAll(CFMENavigateStep):
+    VIEW = VolumeAllView
+    prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
+
+    def step(self, *args, **kwargs):
+        nav = ['Storage', 'Volumes'] if self.obj.appliance.version < '5.8' else \
```

Comment:
I'd advice you to use regular "if" your conditional expression doesn't fit one line. There was a recommendation about that in PEP. Same for list comprehensions.